.changeset/sour-cycles-warn.md

```diff
@@ -0,0 +1,34 @@
+---
+"@latticexyz/cli": patch
+"@latticexyz/store": minor
+"@latticexyz/world": patch
+---
+
+`StoreCore` and `IStore` now expose specific functions for `getStaticField` and `getDynamicField` in addition to the general `getField`.
+Using the specific functions reduces gas overhead because more optimized logic can be executed.
+
+```solidity
+interface IStore {
+  /**
+   * Get a single static field from the given tableId and key tuple, with the given value field layout.
+   * Note: the field value is left-aligned in the returned bytes32, the rest of the word is not zeroed out.
+   * Consumers are expected to truncate the returned value as needed.
+   */
+  function getStaticField(
+    bytes32 tableId,
+    bytes32[] calldata keyTuple,
+    uint8 fieldIndex,
+    FieldLayout fieldLayout
```

Comment:
do you think it would make sense to flip these args?

Reply:
added an issue to get back to this: https://github.com/latticexyz/mud/pull/1521#discussion_r1328005455 (i remember we had talked about moving the `Schema` argument to another location too before it got replaced by `FieldLayout`)